src/app/clusters/resource-monitoring-server/resource-monitoring-server.cpp

```diff
@@ -0,0 +1,300 @@
+/*
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <app-common/zap-generated/attributes/Accessors.h>
+#include <app/AttributePersistenceProvider.h>
+#include <app/InteractionModelEngine.h>
+#include <app/clusters/resource-monitoring-server/resource-monitoring-server.h>
+#include <app/reporting/reporting.h>
+#include <app/util/attribute-storage.h>
+#include <app/util/util.h>
+#include <platform/DiagnosticDataProvider.h>
```

Comment:
Does this file use DiagnosticDataProvider?  I don't think so.

In general, please clean up these includes to include the stuff that should actually be included.  For example, I am pretty sure Accessors.h is not needed, but `ids/Attributes.h` should be included here...

Reply:
rearranged, removed and added needed headers using "include what you use" tool. 
should now include headers that are actually needed and not depend on other headers to include stuff for me.